src/client/dfs/dfs.c

```diff
@@ -4858,6 +4858,13 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 		D_GOTO(out_obj, rc = daos_der2errno(rc));
 	}
 
+	/* If the size was set then do a second stat to fetch the new mtime */
+	if (set_size) {
+		rc = entry_stat(dfs, th, oh, obj->name, len, obj, &rstat, &obj_hlc);
+		if (rc)
+			D_GOTO(out_obj, rc);
+	}
```

Comment:
Is this the right approach, I see many other changes just set rstat.st_mtim = stbuf->st_mtim which would ensure it changed, but then if you called stat again you'd see another change regardless of the fact that only one setattr call was made.

Reply:
is the issue here that you call set_size, but don't get an updated mtime?
in this case, we probably don't even get here since we return here:
https://github.com/daos-stack/daos/pull/9943/files#diff-dd80cb8b1b118e01084005bad5c40fabe2a4b05744db070f7f3dc2142aa98c60R4848

maybe a better approach is that if only set_size is needed to not do the first stat, and just execute a stat after the set_size call, just to avoid 2 stats ?

Reply:
Ah yes, I'd missed that goto.  We need to do the first stat to verify the file hasn't been renamed don't we, but then if we're setting the size we're probably not setting anything else as well so maybe we could remove that.

I'll re-work this but it'll probably be a bigger code change although faster to run.

Reply:
the first entry_stat seems an overkill if we only want to verify that the file has not been renamed. the stat queries the size too which is expensive. so yea we probably want to remove the first entry_stat and maybe replaced it with just a fetch of the entry oid.

Reply:
We want to do both, check that the file is correct but also return full stat information, the current code is efficient where it works as it does both in one request but it doesn't handle this case.